### PR TITLE
ref(docs): Add docstring to `get_crash_frame_from_event_data`

### PR DESCRIPTION
**Pull request by lobsterkatie**
Would love a check that I parsed the logic correctly and didn't miss anything.

### CODE REVIEW

File: src/sentry/stacktraces/processing.py [hunk 1]
```diff
@@ -404,6 +404,19 @@ def process_single_stacktrace(processing_task, stacktrace_info, processable_fram
...
+    Return the highest (closest to the crash) in-app frame in the top stacktrace
+    which doesn't fail the given filter test.
+
+    If no such frame is available, return the highest non-in-app frame which
+    otherwise meets the same criteria.
+
+    Return None if any of the following are true:
+        - there are no frames
+        - all frames fail the given filter test
+        - the data is threaded and there's more than one thread
```
Comment (mitsuhiko) [inline src/sentry/stacktraces/processing.py]: the last one only applies if we can't find an exception

Comment (lobsterkatie) [reply src/sentry/stacktraces/processing.py]: Right. I thought they were mutually exclusive though - you'd have at most one of `exception`, `stacktrace`, and `threads` as a top-level key. Is that not the case?

Comment (lobsterkatie) [reply src/sentry/stacktraces/processing.py]: J/k - just found myself a counterexample. Okay, will fix.

**Comment by lobsterkatie** [comment]
@mitsuhiko I think it should be good now. Lemme know what you think.

**Review by mitsuhiko — APPROVED** [review]

**Pull request merged**